plugins/org.eclipse.epsilon.eol.engine/src/org/eclipse/epsilon/eol/dom/TypeExpression.java

```diff
@@ -61,26 +62,14 @@ else if ("Native".equals(name)) {
 	}
 	
 	@Override
-	public EolType execute(IEolContext context) throws EolRuntimeException {
-		if (type instanceof EolCollectionType) {
-			EolCollectionType collectionType = (EolCollectionType) type;
-
-			// Compute content type if a parameter type has been provided
-			if (parameterTypeExpressions.size() >= 1 && collectionType.getContentType() == EolAnyType.Instance) {
-				collectionType.setContentType(parameterTypeExpressions.get(0).execute(context));
-			}
-		}
-		else if (type instanceof EolMapType) {
-			EolMapType mapType = (EolMapType) type;
-
-			// Compute key type if at least one parameter type has been provided
-			if (parameterTypeExpressions.size() >= 1 && mapType.getKeyType() == EolAnyType.Instance) {
-				mapType.setKeyType(parameterTypeExpressions.get(0).execute(context));
-			}
-
-			// Compute value type if at least two parameter types has been provided
-			if (parameterTypeExpressions.size() >= 2 && mapType.getValueType() == EolAnyType.Instance) {
-				mapType.setValueType(parameterTypeExpressions.get(1).execute(context));
+	public EolType execute(IEolContext context) throws EolRuntimeException {	
```

Comment:
Parameter types should not be resolved during execution; they should be resolved during static analysis.